tenderly/unchainedProxy.ts

```diff
@@ -0,0 +1,144 @@
+/* eslint-disable @shapeshiftoss/logger/no-native-console */
+import type { evm } from '@shapeshiftoss/common-api'
+import dotenv from 'dotenv'
+import express from 'express'
+import { readFileSync } from 'fs'
+import { createProxyMiddleware, fixRequestBody } from 'http-proxy-middleware'
+import Web3 from 'web3'
+import { hexToNumber, numberToHex } from 'web3-utils'
+import type { RawData, WebSocket as WebSocketClient } from 'ws'
+import WebSocket, { WebSocketServer } from 'ws'
+
+import { corsMiddleware } from './middleware/cors'
+import {
+  PROXY_UNCHAINED_ETHEREUM_HTTP_PORT,
+  PROXY_UNCHAINED_ETHEREUM_WS_PORT,
+} from './utils/constants'
+import { getTenderlyRpcUrl } from './utils/getTenderlyRpcUrl'
+import { estimateGas, getBalance, getTransactionByHash, sendTransaction } from './utils/rpcHelpers'
+import type { EstimateGasParams } from './utils/types'
+
+const { REACT_APP_UNCHAINED_ETHEREUM_HTTP_URL, REACT_APP_UNCHAINED_ETHEREUM_WS_URL } = dotenv.parse(
+  readFileSync('.env'),
+)
+
+const httpPort = Number(PROXY_UNCHAINED_ETHEREUM_HTTP_PORT)
+const websocketPort = Number(PROXY_UNCHAINED_ETHEREUM_WS_PORT)
+const provider = new Web3.providers.HttpProvider(getTenderlyRpcUrl())
+let subscriptionDetails: { subscriptionId: string; address: string } | undefined
+let clientWebsocket: WebSocketClient | undefined
+
+const captureSubscriptionDetails = (rawData: RawData) => {
+  const { subscriptionId, method, data } = JSON.parse(rawData.toString())
+
+  if (method !== 'subscribe' || data.topic !== 'txs') return
+
+  subscriptionDetails = { subscriptionId, address: data.addresses[0] }
+}
+
+const sendTxViaWebSocket = async (txId: string): Promise<void> => {
+  const result = await getTransactionByHash(provider, txId)
+
+  const data: evm.Tx = {
+    txid: txId,
+    blockHash: result.blockHash,
+    blockHeight: hexToNumber(result.blockNumber ?? '0x0'),
```

Comment:
Add a comment that this should never happen but it may?

Reply:
Or possibly fall back to the pending block height if we want a paranoia case, which should pretty much be the same